File: Nibbles/Vector.cs
namespace Nibbles
{
    public enum Direction : int
    { 
        North, 
        East,
        South, 
        West, 
    }

    class Vector : Position, IVector
    {
        public Direction Direction { get; set; }

        public void Move()
        {
            var pos = MoveNew();
            SetPosition(pos);
        }

        public IPosition MoveNew()
        {
            return MoveNew(Direction);
        }

        public IPosition MoveNew(Direction direction)
        {
            var pos = new Position{ X = X, Y = Y };
            if(direction == Direction.North)
            {
                pos.X--;
            }else if(direction == Direction.East)
            {
                pos.Y++;
            }else if(direction == Direction.South)
            {
                pos.X++;                
            }else if(direction == Direction.West)
            {
                pos.Y--;
            }
            return pos;
        }
    }
}

File: JNibbles/src/main/java/org/lagash/nibbles/Vector.java
/**
 * Copyright (C) 2017 DiegoG.
 */
package org.lagash.nibbles;

public class Vector extends Position implements IVector {
    Direction direction;

    public Direction getDirection() {
        return direction;
    }

    public void setDirection(Direction value) {
        direction = value;
    }

    public void move() {
        Position pos = (Position)moveNew();
        setPosition(pos);
    }

    public IPosition moveNew() {
        return moveNew(getDirection());
    }

    public IPosition moveNew(Direction direction) {
        Position pos = new Position();
        pos.setX( getX() );
        pos.setY( getY() );
        if(direction == Direction.North) {
            pos.setY( getY()-1 );
        } else if(direction == Direction.East) {
            pos.setX( getX()+1 );
        } else if(direction == Direction.South) {
            pos.setY( getY()+1 );
        } else if(direction == Direction.West) {
            pos.setX( getX()-1 );
        }
        return pos;
    }
}
